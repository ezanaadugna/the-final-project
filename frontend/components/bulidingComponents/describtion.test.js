import React from "react";
import { render } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import { Text } from "react-native";
import DecriptionComponent from "./description";

describe("DecriptionComponent", () => {
  it("renders correctly", () => {
    const decriptionText = "Test description";
    const tree = renderer
      .create(<DecriptionComponent descriptionText={decriptionText} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("displays the correct decription text", () => {
    const decriptionText = "Test decription";
    const { getByText } = render(
      // Use the render function from the testing library
      <DecriptionComponent decriptionText={decriptionText} />
    );
    const decriptionElement = getByText(decriptionText);
    expect(decriptionElement).toBeDefined();
  });
});
