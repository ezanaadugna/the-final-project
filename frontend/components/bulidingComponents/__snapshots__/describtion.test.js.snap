// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DecriptionComponent renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderRadius": 10,
      "elevation": 3,
      "height": "20%",
      "justifyContent": "center",
      "marginTop": 10,
      "padding": 10,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 4,
      "width": "90%",
    }
  }
>
  <Text
    style={
      {
        "color": "black",
        "fontFamily": "Helvetica",
        "fontSize": 13,
      }
    }
  >
     
  </Text>
</View>
`;
